### DISCUSSION TITLE
Use template in custom handler for 404 or 500 pages

### DISCUSSION THREAD

**Discussion by Aju100** [fastapi/fastapi#6619] [Questions] [answered]
How can I use the Jinja2 template to render my custom template using TemplateResponse like the below code in the following sample code of 404 or 500 exception?
```
from fastapi import FastAPI
from fastapi.templating import Jinja2Templates
@app.get('/')
async def home(request: Request):
    return templates.TemplateResponse("general_pages/home.html",{"request":request})

```
In below code, it used HTMLResponse but How I can use for my custom template like flask or django application.
```
from flask import render_template

@app.errorhandler(404)
def page_not_found(e):
    # note that we set the 404 status explicitly
    return render_template('404.html'), 404
````

```
from starlette.applications import Starlette
from starlette.responses import HTMLResponse


HTML_404_PAGE = <h1>Here is 404</h1>
HTML_500_PAGE = <h1>Here is 500</h1>


async def not_found(request, exc):
    return HTMLResponse(content=HTML_404_PAGE, status_code=exc.status_code)

async def server_error(request, exc):
    return HTMLResponse(content=HTML_500_PAGE, status_code=exc.status_code)


exception_handlers = {
    404: not_found,
    500: server_error
}

app = Starlette(routes=routes, exception_handlers=exception_handlers)
```

Reference:
- [https://www.starlette.io/exceptions/](https://www.starlette.io/exceptions/)
- [https://flask.palletsprojects.com/en/1.1.x/patterns/errorpages/](https://flask.palletsprojects.com/en/1.1.x/patterns/errorpages/)

**Comment by Hultner** [comment, marked as answer]
Take a look at the documentation for [overriding the default exception handlers](https://fastapi.tiangolo.com/tutorial/handling-errors/#override-the-default-exception-handlers). I think that should be enough to achieve what you're asking about.

**Comment by tiangolo** [comment]
Thanks for the help here @Hultner ! 👏  🙇 

Thanks for reporting back and closing the issue @Aju100  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
